website/docs/bestpractices/analytics/trino-best-practices.md

```diff
@@ -0,0 +1,642 @@
+- Enhances stability: Appropriate resource allocation reduces the risk of pod evictions and improves overall cluster stability.
+
+
+### AutoScaling Configuration
+
+We recommend implementing [KEDA](https://keda.sh/) for event-driven scaling in Trino clusters to enable dynamic workload management. The combination of KEDA and Karpenter on Amazon EKS creates a powerful autoscaling solution that eliminates scaling challenges. While KEDA manages fine-grained pod scaling based on real-time metrics, Karpenter handles efficient node provisioning. Together, they replace manual scaling processes, delivering both improved performance and cost optimization.
+
+#### Configuring Keda
+- Add Keda helm release to the cluster
+- Add JVM / JMX Exporter configuration to Trino Helm value, enable serviceMonitor
+```
+configProperties: |-
+      hostPort: localhost:{{- .Values.jmx.registryPort }}
+      startDelaySeconds: 0
+      ssl: false
+      lowercaseOutputName: false
+      lowercaseOutputLabelNames: false
+      whitelistObjectNames: ["trino.execution:name=QueryManager","trino.execution:name=SqlTaskManager","trino.execution.executor:name=TaskExecutor","trino.memory:name=ClusterMemoryManager","java.lang:type=Runtime","trino.memory:type=ClusterMemoryPool,name=general","java.lang:type=Memory","trino.memory:type=MemoryPool,name=general"]
+      autoExcludeObjectNameAttributes: true
+      excludeObjectNameAttributes:
+        "java.lang:type=OperatingSystem":
+          - "ObjectName"
+        "java.lang:type=Runtime":
+          - "ClassPath"
+          - "SystemProperties"
+      rules:
+      - pattern: ".*"
+```
+- Deploy KEDA scaledObject for trino, tracking CPU and QueuedQueries. The 'target' CPU percentage here is set at 85%, to leverage the physical cores of Graviton instances.
+```
+triggers:
+  - type: cpu
+    metricType: Utilization
+    metadata:
+      value: '85'  # Target CPU utilization percentage
+  - type: prometheus
+    metricType: Value
+    metadata:
+      serverAddress: http://kube-prometheus-stack-prometheus.kube-prometheus-stack.svc.cluster.local:9090
+      threshold: '1'
+      metricName: queued_queries
+      query: sum by (job) (avg_over_time(trino_execution_QueryManager_QueuedQueries{job="trino"}[1m]))
+```
+
+## File Caching
+
+File caching provides three major benefits to storage systems and query operations. First, it reduces the storage load by preventing repetitive retrievals of the same files, as cached files can be reused across multiple queries on the same worker. Second, it can significantly improve query performance by eliminating repeated network transfers and allowing access to local file copies, particularly beneficial when the original storage is in a different network or region. Finally, it leads to reduced query costs by minimizing network traffic and storage access.
+
+This is a basic configuration for implementing File Caching.
```

Comment:
Can we add Alluxio relevant configuration also in this section?

Reply:
Its there already